### PR TITLE
Review

### PR DESCRIPTION
**Pull request by benwmx**
#### Review the code against the best practices : DRY, YAGNI, KISS

### CODE REVIEW

File: src/index-src.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <script src="https://kit.fontawesome.com/f654a6a150.js" crossorigin="anonymous"></script>
+  <title>To Do List</title>
+</head>
+
+<body>
+  <div id="list-container" class="list-container">
+    <div class="title">
+      Today's To Do
+      <i class="fas fa-sync-alt"></i>
+    </div>
```
Comment (ahzia) [inline src/index-src.html:17]: Use HTML5 semantic tags `<header>`

File: src/index-src.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <script src="https://kit.fontawesome.com/f654a6a150.js" crossorigin="anonymous"></script>
+  <title>To Do List</title>
+</head>
+
+<body>
+  <div id="list-container" class="list-container">
```
Comment (ahzia) [inline src/index-src.html:13]: Use HTML5 semantic tags `<main>`

File: src/index-src.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <div class="title">
+      Today's To Do
+      <i class="fas fa-sync-alt"></i>
+    </div>
+    <div class="add">
+      <input type="text" name="add-task" id="task-desc" placeholder="add your task">
+      <i id="add" class="fas fa-plus"></i>
+    </div>
+    <ul class="list" id="list">
+
```
Comment (ahzia) [inline src/index-src.html:23]: Do not add blank lines, spaces, or indentations without a reason.

File: src/main-src.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+const listDiv = document.getElementById('list');
+const clearAllButton = document.getElementById('clear-list');
+listDiv.addEventListener('click', (event) => {
+  if (event.target !== event.currentTarget) {
+    if (event.target.className === 'check') {
+      tasks = getStorage();
+      updateStatus(tasks, parseInt(event.target.parentElement.parentElement.id, 10), true);
+      showList(tasks);
+      updateStorage(tasks);
```
Comment (ahzia) [inline src/main-src.js:25]: Whenever you are doing an update to your local storage (calling `updateStorage(tasks)`) you also call `showList(tasks)`, as a suggestion you can use/call these functions inside one function to not repeat it each time.

File: src/storage.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { orderTasks, showList, updateStorage } from './task.js';
+
+export const getStorage = () => {
```
Comment (ahzia) [inline src/storage.js:3]: It will be better to call this function `updateLocalStorage()`

File: src/storage.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+export const updateAfterDrag = (container) => {
+  const listOfElements = container.children;
+  const tasks = getStorage();
+  const orderedTasks = [];
+  Array.from(listOfElements).forEach((element) => {
+    const id = parseInt(element.id, 10);
+    const index = tasks.findIndex((task) => task.index === id);
+    orderedTasks.push(tasks[index]);
+  });
+  orderTasks(orderedTasks);
```
Comment (ahzia) [inline src/storage.js:17]: You are calling these functions together many times, In order to not repeating it, you can have a function and call three of these functions inside it.
You can call that function something like `updateTask(task)`

File: src/task.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    task.className = 'task';
+    task.setAttribute('draggable', 'true');
+    task.appendChild(descriptionContainer);
+    task.appendChild(remove);
+    task.appendChild(edit);
+    listDiv.appendChild(task);
+  }
+};
+
+export const updateStorage = (storage) => {
```
Comment (ahzia) [inline src/task.js:42]: Its better to name this function `updateLocalStorage()`

Comment (ahzia) [inline src/task.js:42]: Its better to name this function updateLocalStorage()

**Review by ahzia — COMMENTED** [review]
@benwmx You did a Great Job. 
I didn't found anything that makes your code quality bad, but there are a few suggestions that will make your code cleaner.

- Separate your HTML, CSS, and javascript files into different folders to better manage them.
- In your `Storage.js` file you don't have all your functions related to local Storage, Instead, you have some functions related to tasks, be more careful with how you manage your functions in different files.
Please check the Inline comments as well.

**Comment by benwmx** [comment]
hi, @ahzia thank you for your thorough review, I implemented all the changes that you suggested.

**Review by ahzia — APPROVED** [review]
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**